### PR TITLE
Retrieve subscription stats

### LINKED ISSUE #4: Retrieve subscription stats
Nakadi provides API to get stats of a subscription, the documentation is [here](https://nakadi.io/manual.html#/subscriptions/subscription_id/stats_get).

This ticket is about adding data type and function to support this functionality.

### PR DESCRIPTION
**Pull request by taojang**
closes #4

### CODE REVIEW

File: src/Nakadi/Client.purs [hunk 1]
```diff
@@ -272,3 +274,19 @@ streamSubscriptionEvents sid@(SubscriptionId subId) streamParameters eventHandle
...
+  => SubscriptionId
+  -> m (NakadiResponse (notFound :: E404) (Array SubscriptionEventTypeStats))
+getSubscriptionStats (SubscriptionId subId) = do
+  let url = "/subscriptions/" <> subId <> "/stats"
+  res :: Either Problem {items :: Array SubscriptionEventTypeStats}  <- getRequest url >>= request >>= deserialise
+  stuff <- (res # catchErrors case _ of
+    p @ { status: 401 } -> pure $ lmap e401 res
+    p @ { status: 404 } -> pure $ lmap e404 res
+    p -> unhandled p)
+  pure $ map _.items stuff
```
Comment (taojang) [inline src/Nakadi/Client.purs]: I have no idea how type inference works here. I've tried to do stuff differently but type checker yelled at me.

Comment (taojang) [reply src/Nakadi/Client.purs]: for example this breaks
```purescript
getSubscriptionStats
  :: ∀ r m
   . MonadAsk (Env r) m
  => MonadThrow Error m
  => MonadAff m
  => SubscriptionId
  -> m (NakadiResponse (notFound :: E404) (Array SubscriptionEventTypeStats))
getSubscriptionStats (SubscriptionId subId) = do
  let url = "/subscriptions/" <> subId <> "/stats"
  res :: Either Problem {items :: Array SubscriptionEventTypeStats}  <- getRequest url >>= request >>= deserialise
  let result = map _.items res
  result # catchErrors case _ of
    p @ { status: 401 } -> pure $ lmap e401 res
    p @ { status: 404 } -> pure $ lmap e404 res
    p -> unhandled p
```

Comment (taojang) [reply src/Nakadi/Client.purs]: Simon helped me out. There's `res` hiding in expressions of pattern matching

**Comment by taojang** [comment]
The type of `catchErrors` is bit crazy, I'm leaving it out for now

File: src/Nakadi/Client.purs (hunk 1 above)
Comment (i-am-the-slime) [reply src/Nakadi/Client.purs]: Wow, I did not see this, sorry, and thanks @reactormonk

**Comment by taojang** [comment]
No need for that

**Pull request closed without merging**